### ISSUE TITLE
1828 shows incorrect ownership/float numbers when parring B&Os

### ISSUE THREAD

**Issue by thepackrat** [tobymao/18xx#5057]
After buying any B&O (any one of the privates that grants 30% in a company), the company shows up at the end of the auction asking you to set the par price. At that time, the company shows 1 share owned by you, and "50% to float".

It should show 3 shares owned by you (because you have bought the B&O) and 30% to float, so epople can select a par that lets them buy 3 shares to float, to avoid giving people a feeling of despair when setting the par price.

**Comment by crericha** [comment]
This is true for *all* games that have a private that comes with a director share.

**Comment by benjaminxscott** [comment]
I tried [moving the buy_shares()](https://github.com/tobymao/18xx/compare/master...benjaminxscott:exchange_president_before_par#diff-8b1c776b20fe190442fcbb137476bcd56329cb1b568e0ee20c8d23f4b0b5af4eR745) call up into the base class, but there's some other code that is coupled to that, like setting `corp.ipoed` etc
